Servers/Infrastructure/OJS.Servers.Infrastructure/Listeners/RedisCircuitBreakerListener.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace OJS.Servers.Infrastructure.Listeners;
+
+using Microsoft.Extensions.Logging;
+using Polly;
+using Polly.Telemetry;
+
+internal class RedisCircuitBreakerListener : TelemetryListener
+{
+    private readonly ILogger<ResiliencePipeline> logger;
+    private int timesCircuitHasBeenOpened;
+    private int timesCircuitHasBeenClosed;
+    private int timesCircuitHasBeenHalfOpen;
+    private int totalNumberOfRetries;
+    private int totalNumberOfCacheGetCalls;
+
+    public RedisCircuitBreakerListener(ILogger<ResiliencePipeline> logger)
+        => this.logger = logger;
+
+    public override void Write<TResult, TArgs>(in TelemetryEventArguments<TResult, TArgs> args)
+    {
+        var operationName = args.Context.Properties.GetValue(new ResiliencePropertyKey<string>("OperationKey"), "This value should not be null. Use an appropriate operation key!");
+
+        var logMessage = args.Event.EventName switch
+        {
+            "OnCircuitOpened" => $"Circuit breaker opened. Total number of times opened: {++this.timesCircuitHasBeenOpened}. Event: {args.Event}. Outcome: {(args.Outcome.HasValue ? args.Outcome.Value : "null")} Pipeline: {args.Source.PipelineName}. Strategy: {args.Source.StrategyName}.",
+            "OnCircuitClosed" => $"Circuit breaker closed. Total number of times closed: {++this.timesCircuitHasBeenClosed}. Event: {args.Event}. Outcome: {(args.Outcome.HasValue ? args.Outcome.Value : "null")} Pipeline: {args.Source.PipelineName}. Strategy: {args.Source.StrategyName}.",
```

Comment:
Also here. You can directly log the message with template syntaxis, instead of doing all the checks after this.

Reply:
Done.